Orthtree/include/CGAL/Orthtree.h

```diff
@@ -538,77 +629,131 @@ class Orthtree
+      Local_coordinates local_coords;
+      std::size_t dim = 0;
       for (const auto& r : cartesian_range(center, point))
-        index[dimension ++] = (get<0>(r) < get<1>(r));
+        local_coords[dim++] = (get<0>(r) <= get<1>(r));
 
       // Find the correct sub-node of the current node
-      node_for_point = node_for_point[index.to_ulong()];
+      node_for_point = child(node_for_point, local_coords.to_ulong());
     }
 
     // Return the result
     return node_for_point;
   }
 
   /*!
-    \brief finds the `k` nearest neighbors of `query`.
+  \brief finds the `k` nearest neighbors of the point `query`.
 
-    Nearest neighbors are outputted in order of increasing distance to
-    `query`.
+  Nearest neighbors are outputted in order of increasing distance to `query`.
 
-    \tparam OutputIterator a model of `OutputIterator` that accept `Point_d` objects.
-    \param query a query point.
-    \param k the number of neighbors.
-    \param output the output iterator.
-   */
+  \tparam OutputIterator a model of `OutputIterator` that accepts `GeomTraits::Node_data_element` objects.
+
+  \param query query point
+  \param k number of neighbors to find
+  \param output output iterator
+
+  \warning Nearest neighbor searches requires `GeomTraits` to be a model of `CollectionPartitioningOrthtreeTraits`.
+ */
   template<typename OutputIterator>
-  OutputIterator nearest_neighbors (const Point& query,
-                                    std::size_t k,
-                                    OutputIterator output) const {
-    Sphere query_sphere (query, (std::numeric_limits<FT>::max)());
+  auto nearest_neighbors(const Point& query,
+    std::size_t k,
+    OutputIterator output) const -> std::enable_if_t<supports_neighbor_search, OutputIterator> {
+    Sphere query_sphere(query, (std::numeric_limits<FT>::max)());
+    CGAL_precondition(k > 0);
+
     return nearest_k_neighbors_in_radius(query_sphere, k, output);
   }
 
   /*!
-    \brief finds the points in `sphere`.
+  \brief finds the elements in the sphere `query`.
 
-    Nearest neighbors are outputted in order of increasing distance to
-    the center of `sphere`.
+  Elements are outputted in order of increasing distance to
+  the center of the sphere.
 
-    \tparam OutputIterator a model of `OutputIterator` that accept `Point_d` objects.
-    \param query query sphere.
-    \param output output iterator.
-   */
+  \tparam OutputIterator a model of `OutputIterator` that accepts `GeomTraits::Node_data_element` objects.
+
+  \param query query sphere
+  \param output output iterator
+
+  \warning Nearest neighbor searches requires `GeomTraits` to be a model of `CollectionPartitioningOrthtreeTraits`.
+ */
   template<typename OutputIterator>
-  OutputIterator nearest_neighbors (const Sphere& query, OutputIterator output) const {
+  auto neighbors_in_radius(const Sphere& query, OutputIterator output) const -> std::enable_if_t<supports_neighbor_search, OutputIterator> {
```

Comment:
In Kd-tree this function is called [`search()`](https://doc.cgal.org/latest/Spatial_searching/classCGAL_1_1Kd__tree.html#a4e760af1ae3da968dec45b378db207fc) and they also have `search_any()`

Reply:
I think search could be confused with locate? I would just add _k_ to the nearest neighbor function without a radius to make it clear.